safenode/src/domain/wallet/local_store.rs

```diff
@@ -22,49 +24,69 @@ use std::{
     path::{Path, PathBuf},
 };
 
+const WALLET_DIR_NAME: &str = "wallet";
+
 /// A wallet that can only receive tokens.
 pub struct LocalWallet {
     /// The secret key with which we can access
     /// all the tokens in the available_dbcs.
     key: MainKey,
     /// The wallet containing all data.
     wallet: KeyLessWallet,
-    /// The dir of the wallet file.
-    root_dir: PathBuf,
+    /// The dir of the wallet file, main key, public address, and new dbcs.
+    wallet_dir: PathBuf,
 }
 
 impl LocalWallet {
     /// Stores the wallet to disk.
     pub async fn store(&self) -> Result<()> {
-        store_wallet(&self.root_dir, &self.wallet).await
+        store_wallet(&self.wallet_dir, &self.wallet).await
+    }
+
+    /// Stores the given dbc to the `created dbcs dir` in the wallet dir.
+    /// Each recipient has their own dir, containing all dbcs for them.
+    /// These can then be sent to the recipients out of band, over any channel preferred.
+    pub async fn store_created_dbc(&mut self, dbc: Dbc) -> Result<()> {
+        store_created_dbcs(vec![dbc], &self.wallet_dir).await
```

Comment:
Just to help me understand, are the wallet operations all local? So when you run the `wallet send` command as a user, you're not actually sending anything to someone else, you just create a local directory for the recipient where the DBC is saved? So how do you actually perform the send for real?

Reply:
In the units its all local like you describe, because we have not injected a connected client, just a mock.

But the implementation in the safe client injects an actual connected client, so it gets uploaded to network on send 👍

Reply:
Ah, I think I now read your question more as it was intended..

The actual dbcs... yes how do they reach the recipient. It's exactly as you describe, it goes to the created_dbcs directory in the client wallet path.

The user has to deliver it out of band to the recipient. On paper, on a thumb drive, in email, on whatsapp. What ever suits :)